### PR TITLE
Upgrade to Scio 0.10.4 + beam 2.30.0

### PR DESCRIPTION
**Pull request by aherbst-broad**
We need to upgrade to at least beam 2.30.0 to get dataflow runner v2 (which we need to be able to use a custom container in order to specify UTF-8 file encodings for msgpack encoding).

This PR upgrades Scio + Beam and adds a nowarn on the caseapp arg parsing deprecation which is now popping up.

### CODE REVIEW

File: core-sbt-plugins/src/main/scala/org/broadinstitute/monster/sbt/core/MonsterBasePlugin.scala [hunk 1]
```diff
@@ -62,7 +62,7 @@ object MonsterBasePlugin extends AutoPlugin {
   override def buildSettings: Seq[Def.Setting[_]] =
     Seq(
       organization := "org.broadinstitute.monster",
-      scalaVersion := "2.12.11",
+      scalaVersion := "2.12.15",
```
Comment (aherbst-broad) [inline core-sbt-plugins/src/main/scala/org/broadinstitute/monster/sbt/core/MonsterBasePlugin.scala:65]: We need scala 2.12.15 to get the @nowarn  annotation

File: core-sbt-plugins/src/main/scala/org/broadinstitute/monster/sbt/core/MonsterLibraryPlugin.scala [hunk 2]
```diff
@@ -59,6 +59,7 @@ object MonsterLibraryPlugin extends AutoPlugin {
 
   override def projectSettings: Seq[Def.Setting[_]] =
     Seq(
+      resolvers += "Confluent" at "https://packages.confluent.io/maven/",
```
Comment (aherbst-broad) [inline core-sbt-plugins/src/main/scala/org/broadinstitute/monster/sbt/core/MonsterLibraryPlugin.scala:62]: Confluent ilbs are now externalized in this version of scio/beam.

File: core-sbt-plugins/src/main/scala/org/broadinstitute/monster/sbt/core/MonsterBasePlugin.scala (hunk 1 above)
Comment (aherbst-broad) [inline core-sbt-plugins/src/main/scala/org/broadinstitute/monster/sbt/core/MonsterBasePlugin.scala:63]: We need scala 2.12.15 to get the `@nowarn` annotation

**Review by quazi-h — APPROVED** [review]
Changes look good

**Pull request closed without merging**